main.js

```diff
@@ -139,13 +147,46 @@ const yaml = require('js-yaml');
       const paths = await globber.glob();
 
       await Promise.all(
-        paths.map((path) =>
-          exec.exec(
+        paths.map( async (path) => {
+          let result = await exec.getExecOutput(
             'charmcraft',
             ['upload', '--quiet', '--release', channel, path].concat(revisions)
-          )
-        )
+          );
+	  charmRevision = result.stdout.split(' ')[1];
+	})
       );
+
+      let githubToken = core.getInput('github-token');
+      if (!githubToken) {
+        if (process.env.GITHUB_TOKEN) {
+          githubToken = process.env.GITHUB_TOKEN;
+        } else {
+          core.setFailed(
+            'Input "github-token" is missing, and not provided in environment'
+          );
+        }
+      }
+
+      const octokit = github.getOctokit(githubToken);
+      const tagName = `${track}-rev${charmRevision}`;
+      var tagMessage = `${resourceInfo}channel: ${channel}`;
```

Comment:
I don't believe we can handle `track`/`channel` the same as we can `charmRevision` or resource revisions.  The revisions are immutable facts, whereas the track/channel is a property that may be added or removed.  I think these should be separate tags.  

For `track` and `channel`, these will be added and subtracted as things change in charmhub.  This action needs to handle that as well.  So when this action creates a `channel` tag of `latest/stable`, it also needs to remove the `latest/stable` tag from whatever other commit has it now.  This issue is visible in the dummy repo tags [latest-rev21](https://github.com/DomFleischmann/doms-dummy-charm/releases/tag/latest-rev21) and [latest-rev22](https://github.com/DomFleischmann/doms-dummy-charm/releases/tag/latest-rev22), which both assert they're in `latest/edge`

Reply:
I wouldn't bother trying to attach the `latest/stable`, `latest/edge` etc. to the tags. Charmhub clearly shows which revisions are associated with each channel, so I think tagging with just `rev21` is fine. 

The tag message could be amended each time a given revision is posted to a channel?

Reply:
And by that I mean, when something is released to a channel, append a line like the following to the tag message:

```
Released to `latest/stable` at 14:59:45 UTC on 08 Feb 2022
```